doc/btp_spec.txt

```diff
@@ -3760,3 +3772,718 @@ Commands and responses:
+					Address (6 octets)
+					Status (1 octet)
+					Characteristics_Count (1 octet)
+					[array] Characteristic (variable)
+
+		Object Characteristic is defined as:
+					Characteristic_Handle (2 octets)
+					Value_Handle (2 octets)
+					Properties (1 octet)
+					UUID_Length (1 octet)
+					UUID (2 or 16 octets)
+
+		Valid UUID_Length parameter values:
+					0x02 = UUID16
+					0x10 = UUID128
+
+		If Status is non-zero Characteristics_Count shall be set to 0.
+
+	Opcode 0x85 - Discover Characteristics by UUID Response
+
+		Controller Index:	<controller id>
+		Event parameters:	Address_Type (1 octet)
+					Address (6 octets)
+					Status (1 octet)
+					Characteristics_Count (1 octet)
+					[array] Characteristic (variable)
+
+		Object Characteristic is defined as:
+					Characteristic_Handle (2 octets)
+					Value_Handle (2 octets)
+					Properties (1 octet)
+					UUID_Length (1 octet)
+					UUID (2 or 16 octets)
+
+		Valid UUID_Length parameter values:
+					0x02 = UUID16
+					0x10 = UUID128
+
+		If Status is non-zero Characteristics_Count shall be set to 0.
+
+	Opcode 0x86 - Discover All Characteristic Descriptors Response
+
+		Controller Index:	<controller id>
+		Event parameters:	Address_Type (1 octet)
+					Address (6 octets)
+					Status (1 octet)
+					Descriptors_Count (1 octet)
+					[array] Descriptor (variable)
+
+		Object Descriptor is defined as:
+					Descriptor_Handle (2 octets)
+					UUID_Length (1 octet)
+					UUID (2 or 16 octets)
+
+		Valid UUID_Length parameter values:
+					0x02 = UUID16
+					0x10 = UUID128
+
+		If Status is non-zero Descriptors_Count shall be set to 0.
+
+	Opcode 0x87 - Read Characteristic Value/Descriptor Response
+
+		Controller Index:	<controller id>
+		Event parameters:	Address_Type (1 octet)
+					Address (6 octets)
+					Status (1 octet)
+					Data_Length (2 octets)
+					Data (variable)
+
+		If Status is non-zero Data_Length shall be set to 0.
+
+	Opcode 0x88 - Read Using Characteristic UUID Response
+
+		Controller Index:	<controller id>
+		Event parameters:	Address_Type (1 octet)
+					Address (6 octets)
+					Status (1 octet)
+					Data_Length (2 octets)
+					Value_Len (1 octet)
+					Data (variable)
+
+		Data is array of handle-value pairs defined as:
+					Handle (2 octets)
+					Value (Value_Len octets)
+
+		If Status is non-zero Data_Length shall be set to 0.
+
+	Opcode 0x89 - Read Long Characteristic Value/Descriptor Response
+
+		Controller Index:	<controller id>
+		Event parameters:	Address_Type (1 octet)
+					Address (6 octets)
+					Status (1 octet)
+					Data_Length (2 octets)
+					Data (variable)
+
+		If Status is non-zero Data_Length shall be set to 0.
+
+	Opcode 0x8a - Read Multiple Characteristic Values Response
+
+		Controller Index:	<controller id>
+		Event parameters:	Address_Type (1 octet)
+					Address (6 octets)
+					Status (1 octet)
+					Data_Length (2 octets)
+					Data (variable)
+
+		If Status is non-zero Data_Length shall be set to 0.
```

Comment:
As we are sending multiple responses of this kind, I suggest that we define that last one has `Status=0` and `Data_Length=0`

Reply:
why multiple responses?  it should be single response where data is set of two (or more) values